### PR TITLE
IDE-1002 Fixed workspace crash

### PR DESCRIPTION
**Pull request by dehilsterlexis**
Also fixed a related label problem revealed with the crash fix.

### CODE REVIEW

**Comment by GordonSmith** [comment]
@dehilsterlexis looks like this needs to be rebased?

File: eclide/mainfrm.cpp [hunk 1]
```diff
@@ -2531,11 +2537,11 @@ void CMainFrame::DoWorkspaceLoad(IWorkspace * workspace)
...
-    workspace->GetChildWorkspaces(workspaceItems);
-    for(IWorkspaceVector::const_iterator itr = workspaceItems.begin(); itr != workspaceItems.end(); ++itr)
-    {
-        DoWorkspaceLoad(itr->get());
-    }
+    //workspace->GetChildWorkspaces(workspaceItems);
+    //for(IWorkspaceVector::const_iterator itr = workspaceItems.begin(); itr != workspaceItems.end(); ++itr)
+    //{
+    //    DoWorkspaceLoad(itr->get());
+    //}
```
Comment (GordonSmith) [inline eclide/mainfrm.cpp:2544]: Commented code

**Comment by dehilsterlexis** [comment]
@GordonSmith did that.

File: eclide/mainfrm.cpp (hunk 1 above)
Comment (dehilsterlexis) [reply eclide/mainfrm.cpp:2544]: Yes. I just saw that. Happened when I rebased. Will look at that.

**Review by GordonSmith — APPROVED** [review]
Looks good (once the commented code is removed)

**Comment by dehilsterlexis** [comment]
@GordonSmith comments removed.

File: comms/WorkspaceItem.h [hunk 2]
```diff
@@ -27,6 +27,7 @@ __interface __declspec(uuid("A4A4FFF2-0DED-422F-B06B-B0FE4CDB6C52")) IWorkspaceI
     const TCHAR *GetChecksum() const;
 
     void SetContent(const CPersistMap & pm);
+    void SetLabel(const std::_tstring label);
```
Comment (GordonSmith) [inline comms/WorkspaceItem.h:30]: Should be `contst std::_tstring & label`

File: comms/WorkspaceItem.cpp [hunk 3]
```diff
@@ -399,6 +405,11 @@ class CWorkspaceItem : public IWorkspaceItem, public clib::CLockableUnknown
         m_attribute = attribute;
     }
 
+    void SetLabel(const std::_tstring label)
```
Comment (GordonSmith) [inline comms/WorkspaceItem.cpp:408]: Should be `const std::_tstring & label`

**Review by GordonSmith — CHANGES_REQUESTED** [review]

File: comms/WorkspaceItem.h (hunk 2 above)
Comment (dehilsterlexis) [reply comms/WorkspaceItem.h]: Done

File: comms/WorkspaceItem.cpp (hunk 3 above)
Comment (dehilsterlexis) [reply comms/WorkspaceItem.cpp]: Done

**Comment by dehilsterlexis** [comment]
@GordonSmith for review

**Review by GordonSmith — APPROVED** [review]
@richardkchapman ready to pull

**Pull request merged**